Beistellung_Implementierung/system/Coding/src/app/State/SetParameters.c

```diff
@@ -0,0 +1,56 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2024   -   www.newtec.de
+***************************************************************************************************/
+/**
+ * @file       SetParameters.c
+ *
+ *    Module handles setting of parametersets and returns the currently selected parameter set on request
+ */
+/**************************************************************************************************/
+
+/* INCLUDES ***************************************************************************************/
+#include "SetParameters.h"
+
+#include "service/Display.h"
+#include "os/SoftTimer.h"
+#include "app/EventEnum.h"
+
+/* CONSTANTS **************************************************************************************/
+#define OFFSET_FOR_CHAR (48)
+#define ARRAY_SIZE (3)
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+
+/* VARIABLES **************************************************************************************/
+static UInt8 gParamSetIdx = 0;           /**< internal parametersetindex*/
+static ParamSet gParamSetArray[ARRAY_SIZE] = {{255, 5, 5, 5 ,5}, 
+{150, 3, 3, 3, 3}, {75, 1, 1, 1, 1}};    /**< array of parameterset, access via gParamSetIdx*/
```

Comment:
as a tip, since you probably don't want any functions to change this during runtime, you could declare them as const to prevent accidental overwrites

Reply:
(We already made a lot of changes to the code of the implementation phase for a running final version. Therefore changes won't be adresses directly in this old version but in the new one.)

In the new version there are actually allowed and wanted write accesses to this variable (but only in special cases).